README.md

```diff
@@ -1 +1,189 @@
+# │       (1): "1,2.0"
+# └       (3): "5.0,6"
+```
+Note that the "Row count by status" uses the `RowStatus` information we store in `TaskResultBuffer`s to show that there were 3 (`✓`) rows that were parsed successfully, 2 rows have missing values for at least one field (`?`) and 2 rows had at least one parsing error (`!`). 
+See the docstring for `TaskResultBuffer` for more information.
+
+`parse_file` also accepts a number of keyword arguments that can be used to tweak the behavior of the parser by customizing how `Parsers.jl` is used internally.
+
+```julia
+parse_file(
+    IOBuffer("""
+        # a comment
+        i;f;s
+        1_000;2,99;"A string"
+        2_000;4,99;"A string with an \\\"escaped\\\" quote"
+        3_000_000;6,99;"Usually, you'd use \\\" for the escapechar"
+        40;8,99;"But not in this example"
+        5_000_000;10,99;"No sir"
+        """
+    ),
+    [Int, Float64, String],
+    delim=';',
+    decimal=',',
+    groupmark='_',
+    escapechar='\\',
+    comment='#',
+)
+# [ Info: Start row: 1, nrows: 5, Task (runnable) @0x000000010931c010 ❚
+```
+See the docstring for more details.
+
+### Example: Custom `AbstractConsumeContext` for interactive use
+
+It could be handy to have the option to provide an anonymous function to be called on each chunk. 
+We can do that by creating a custom `AbstractConsumeContext` subtype that stores a function and dispatching on it in the `consume!` method.
+
+```julia
+using ChunkedCSV
+
+struct ClosureContext{F} <: AbstractConsumeContext
+    f::F
+end
+ChunkedCSV.consume!(ctx::ClosureContext, payload::ParsedPayload) = ctx.f(payload)
+```
+For increased ergonomics, we can also define a convenience method that calls `parse_file` with the custom context using the `do` syntax.
+
+```julia
+ChunkedCSV.parse_file(f::Function, input, schema; kwargs...) = parse_file(input, schema, ClosureContext(f); kwargs...)
+
+io = IOBuffer("""
+    a,b
+    1,2
+    3,4
+    5,6
+    7,8
+    9,10
+    """
+)
+
+parse_file(io, [Int, Int], buffersize=8) do payload
+    println("Hi there! We're at start row: $(payload.row_num), nrows: $(payload.len) in this chunk")
+end
+# Hi there! We're at start row: 1, nrows: 1 in this chunk
+# Hi there! We're at start row: 2, nrows: 2 in this chunk
+# Hi there! We're at start row: 4, nrows: 1 in this chunk
+# Hi there! We're at start row: 5, nrows: 1 in this chunk
+```
+
+## Notable differences to other CSV parsers
+
+When compared to established packages like [`CSV.jl`](https://github.com/JuliaData/CSV.jl), our package has a few notable differences:
+* We don't do any type inference for column types, so the user has to provide a schema. Otherwise, all columns will be parsed as `String`s.
+* We don't give you back any table-like object. Instead all, results are only temporarily available in the `consume!` method of the `AbstractConsumeContext` you provide. You could use this to build a table-like object inside of the context object, but we don't do it for you.
```

Comment:
I like this section a lot! Presumably there are some positives you could put in here, like "we fully utilize available threads to achieve higher throughput"? or something like that?